### PR TITLE
Me: 2fa Reauth Required Improvements

### LINKED ISSUE #400: Me: 2FA: Improve the 2FA popup dialog
Currently when your 2FA cookie expires you see a dialog asking you to verify your login with the six digit code. The placeholder on the input displays '123456' which could confuse users. Also the submit button should be disabled until there is something in the verification field.

Before:
screen shot 2015-06-30 at 10 42 35 am

Ideally:
screen shot 2015-06-30 at 10 44 08 am

Suggestion:
The text could also be greyed out a bit more to indicate it's a placeholder.

### LINKED ISSUE #401: Me: focus on two-step response field when authenticating access
If you have to authenticate your session before getting access to /me, we might as well focus that field. This is especially nice on mobile where otherwise it may require another tap in the middle of a potentially large device to activate the field. When all you really want to do is enter the code.

Current:
screen shot 2015-06-09 at 04 32 53

Suggested:
screen shot 2015-06-09 at 04 33 36

### PR DESCRIPTION
**Pull request by ebinnion**
Fixes #401
Fixes #400

This PR allows auto focusing of the 2fa code input when the reauth required dialog opens in `/me`.

reauth-autofocus

cc @v18 and @lezama for review

To test:
- Checkout `fix/me-reauth-autofocus`
- Make sure to test with an account that has 2fa enabled
- Go to WordPress.com
- Delete `twostep_auth` cookie
- Go to `/me` in Calypso
- In the dialog that pops up, the input should focus automagically

### CODE REVIEW

**Comment by rickybanister** [comment]
While you're in there, what about addressing #400 

* using mid-points instead of `123456` for the placeholder
* disabling the submit button if the field is empty

**Comment by ebinnion** [comment]
I just pushed up a commit that addresses disabling the "Verify" button when a code is not entered. I'm not sure what we're going to do with the placeholder text just yet, so I'd prefer to handle that in another PR.

reauth-verify-focus

**Comment by ebinnion** [comment]
After some clarification in Slack, I went ahead and updated the placeholder text to be `e.g. 123456`:

screen shot

File: client/me/reauth-required/index.jsx [hunk 1]
```diff
@@ -148,14 +152,21 @@ module.exports = React.createClass( {
...
-			codePlaceholder = '1234567';
+			codePlaceholder = this.translate( 'e.g. 1234567', {
+				context: '7 digit two factor code placeholder.',
+				textOnly: true
+			} );
 		}
 
 		return (
 			<Dialog
+				autoFocus={ false }
```
Comment (lezama) [inline client/me/reauth-required/index.jsx]: Is this prop necessary?

Comment (ebinnion) [inline client/me/reauth-required/index.jsx]: In my testing it is necessary.

My understanding is the the `Dialog` component auto focuses on the content by default. So passing in `autoFocus={false}` short circuits that and allows us to set our own `autoFocus`.

See [this block](https://github.com/Automattic/wp-calypso/blob/master/client/components/dialog/dialog-base.jsx#L36-L38).

**Comment by rickybanister** [comment]
Looks good to me design-wise.

**Comment by lezama** [comment]
Works as expected :ship:

**Pull request merged**